contracts/base/CruizeVault.sol

```diff
@@ -124,84 +143,186 @@ contract CruizeVault is ReentrancyGuardUpgradeable, Module {
+
     /**
-     * @notice This function will handle ETH deposits and.
-     * mint crTokens against the deposited amount in 1:1.
+     * @notice  Deposits the `asset` from msg.sender.
      * @param _amount user depositing amount.
      */
-    function _depositETH(uint256 _amount) internal {
-        if (_amount == 0) revert ZeroAmount(_amount);
+    function _depositETH(uint256 _amount) internal numberIsNotZero(_amount) {
         _updateDepositInfo(ETH, _amount);
         if (gnosisSafe.balance == vaults[ETH].cap)
             revert TokenReachedDepositLimit(vaults[ETH].cap);
+        // transfer token to gnosis valut.
         (bool sent, ) = gnosisSafe.call{value: _amount}("");
-        require(sent, "Failed to transfer Ether");
+        if (!sent) revert FailedToTransferETH();
         emit Deposit(msg.sender, _amount, ETH);
     }
 
     /**
-     * @notice This function will handle ERC20 deposits and.
-     * mint crTokens against the deposited amount in 1:1.
+     * @notice Deposits the `asset` from msg.sender
      * @param _token depositing token address.
-     * @param _amount user depositing amount.
+     * @param _amount is the amount of `asset` to deposit.
      */
-    function _depositERC20(address _token, uint256 _amount) internal {
-        if (_token == address(0)) revert ZeroAddress(_token);
-        if (_amount == 0) revert ZeroAmount(_amount);
-        if (cruizeTokens[_token] == address(0)) revert AssetNotAllowed(_token);
-        IERC20 token = IERC20(_token);
-        if (token.balanceOf(gnosisSafe) == vaults[_token].cap)
+    function _depositERC20(address _token, uint256 _amount)
+        internal
+        tokenIsAllowed(_token)
+        numberIsNotZero(_amount)
+    {
+        if (IERC20(_token).balanceOf(gnosisSafe) == vaults[_token].cap)
             revert TokenReachedDepositLimit(vaults[_token].cap);
         _updateDepositInfo(_token, _amount);
+        // transfer token to gnosis vault.
         IERC20(_token).safeTransferFrom(msg.sender, gnosisSafe, _amount);
         emit Deposit(msg.sender, _amount, _token);
     }
 
     /**
-     * @notice This function will handle instant withdrawals.
-     * i.e if user deposit in 100 round and want to withdraw
-     * in the same round then "withdrawInstantly" will transfer
-     * user funds from Gnosis Safe to user address.
-     * @param _amount user withdrawal amount.
+     * @notice Deposits the `asset` from msg.sender added to `msg.sender`'s deposi receipts.
+     * @notice Used for vault -> vault deposits on the user's behalf.
+     * @param _amount is the amount of `asset` to deposit.
+     * @param _token  is the deposits `asset` address.
+     */
+    function _updateDepositInfo(address _token, uint256 _amount) private {
+        Types.VaultState storage vaultState = vaults[_token];
+        uint16 currentRound = vaultState.round;
+        uint256 decimal = decimals(_token);
+
+        Types.DepositReceipt storage depositReceipt = depositReceipts[
```

Comment:
use memory storage because we are not writing in the state variable
`Types.DepositReceipt memory depositReceipt`